### PR TITLE
Modal window

### PR DESCRIPTION
**Pull request by KekStroke**

### CODE REVIEW

File: src/components/Bracket/BracketMatch.tsx [hunk 1]
```diff
@@ -56,17 +60,31 @@ const BracketMatch = ({
...
+                    firstPlayerName: firstPlayer.name,
+                    secondPlayerName: secondPlayer.name,
+                  }),
+                )
+                  .then((res) => {
+                    const score = res.payload as ModalOutput;
+                    dispatch(finishMatch({ matchId, ...score }));
+                  })
+                  .catch((err) => {
+                    // eslint-disable-next-line no-console
```
Comment (AhmadAlhussin2) [inline src/components/Bracket/BracketMatch.tsx:80]: It is better to mark at as a "TODO: handle errors".

File: src/redux/slices/modal/modalSlice.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+          event.target.removeEventListener("submit", handleSubmit);
+        };
+
+        waitForElement("#formModal")
+          .then(() => {
+            const form = document.getElementById("formModal");
+            form?.addEventListener("submit", handleSubmit);
+          })
+          .catch((err) => {
+            // eslint-disable-next-line no-console
```
Comment (AhmadAlhussin2) [inline src/redux/slices/modal/modalSlice.ts:42]: Same, it is better to make at as a todo

File: src/redux/slices/bracket/bracketSlice.ts [hunk 3]
```diff
@@ -28,14 +28,22 @@ export const bracketSlice = createSlice({
...
-    finishMatch: (state, action: PayloadAction<{ matchId: number }>) => {
+    finishMatch: (
+      state,
+      action: PayloadAction<{
+        matchId: number;
+        firstPlayerScore: number;
+        secondPlayerScore: number;
+      }>,
+    ) => {
       //	TODO: Check if match was initialized
```
Comment (AhmadAlhussin2) [inline src/redux/slices/bracket/bracketSlice.ts:39]: Is it enough to check if `isStarted is true`?

Comment (KekStroke) [reply src/redux/slices/bracket/bracketSlice.ts:39]: We are not given a match here. Only matchId. This is a for a function to set scores to a match with a certain id

File: src/redux/slices/modal/modalSlice.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            console.error(err);
+          });
```
Comment (KekStroke) [inline src/redux/slices/modal/modalSlice.ts:44]:
```suggestion
          .catch((err) => {
            // TODO: handle errors
            // eslint-disable-next-line no-console
            console.error(err);
          });
```

**Pull request merged**